Frameworks/Foundation/NSMutableSet.mm

```diff
@@ -139,11 +139,21 @@ - (void)removeObject:(id)object {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (void)filterUsingPredicate:(NSPredicate*)predicate {
-    UNIMPLEMENTED();
+    if (predicate == NULL) {
+        return;
+    }
+    @synchronized(self) {
+        NSArray* objects = [self allObjects];
+
+        for (id obj in objects) {
+            if (![predicate evaluateWithObject:obj]) {
+                [self removeObject:obj];
```

Comment:
Nevermind: you're using `-allObjects`.

---
In reply to: [74296016](https://github.com/Microsoft/WinObjC/pull/857#discussion_r74296016) [](ancestors = 74296016)